language/src/regent/codegen.t

```diff
@@ -3810,6 +3827,24 @@ function codegen.expr_cross_product_array(cx, node)
     expr_type)
 end
 
+-- Returns Terra expression for subregion of partition of specified color.
+-- The color space can be either structured or unstructured, i.e. `color` can
+-- be an integer or of an index type.
+local function get_partition_subregion(cx, partition, color)
```

Comment:
Implicit-cast color to partition:colors().index_type, then call to_domain_point.